### PR TITLE
Built in React support in route objects

### PR DESCRIPTION
**Pull request by garrensmith**

### CODE REVIEW

File: app/addons/cors/components.react.jsx [hunk 1]
```diff
@@ -225,7 +225,8 @@ define([
         origins: corsStore.getOrigins(),
         isAllOrigins: corsStore.isAllOrigins(),
         configChanged: corsStore.hasConfigChanged(),
-        savingStatus: corsStore.getSavingStatus()
+        shouldSaveChange: corsStore.shouldSaveChange()
+        //savingStatus: corsStore.getSavingStatus()
```
Comment (robertkowalski) [inline app/addons/cors/components.react.jsx]: do we still need that?

Comment (robertkowalski) [inline app/addons/cors/components.react.jsx]: if not we can also delete https://github.com/apache/couchdb-fauxton/blob/9343084ace991386fab014529efdfdc67e10f389/app/addons/cors/stores.js#L35 et al (setter, actiontype...)

File: app/addons/documents/tests/nightwatch/paginateView.js [hunk 2]
```diff
@@ -13,7 +13,7 @@
 module.exports = {
 
 	'change number of items per page': function (client) {
-    var waitTime = client.globals.maxWaitTime,
+    var waitTime = 60000, //client.globals.maxWaitTime,
```
Comment (robertkowalski) [inline app/addons/documents/tests/nightwatch/paginateView.js]: i think this was used for debugging

File: app/addons/documents/routes-index-editor.js [hunk 3]
```diff
@@ -52,7 +54,7 @@ function (app, FauxtonAPI, Helpers, BaseRoute, Documents, Index,
 
     establish: function () {
       return [
-        this.designDocs.fetch({reset: true}),
+        //this.designDocs.fetch({reset: true}),
```
Comment (robertkowalski) [inline app/addons/documents/routes-index-editor.js]: we can delete that one

File: app/core/layout.js [hunk 4]
```diff
@@ -11,9 +11,10 @@
 // the License.
 
 define([
-  "backbone",
-  "plugins/backbone.layoutmanager"
-], function (Backbone) {
+  'backbone',
+  'react',
+  'plugins/backbone.layoutmanager'
+], function (Backbone, React) {
```
Comment (robertkowalski) [inline app/core/layout.js]: i think react is not used as a dep in this file

File: app/addons/documents/routes-documents.js [hunk 5]
```diff
@@ -188,33 +187,9 @@ function (app, FauxtonAPI, BaseRoute, Documents, Changes, Index, DocEditor, Mang
...
-        if (this.sidebar) {
-          this.removeView('#sidebar');
-        }
-        if (this.footer) {
-          this.removeView('#footer');
-        }
-        if (this.headerView) {
-          this.removeView('#dashboard-upper-content');
-        }
-
```
Comment (robertkowalski) [inline app/addons/documents/routes-documents.js]: :heart_eyes:

File: app/core/routeObject.js [hunk 6]
```diff
@@ -229,10 +239,34 @@ function (FauxtonAPI, Backbone) {
...
+    },
+
+    removeComponents: function () {
+      _.each(this.reactComponents, function (component, selector) {
+        this.removeComponent(selector);
+      }, this);
+
+      this.reactComponents = {};
+    },
+
```
Comment (robertkowalski) [inline app/core/routeObject.js]: can you write a few unit tests for those methods?

You will probably need https://github.com/apache/couchdb-fauxton/pull/356 before, going to merge it now

**Comment by benkeen** [comment]
This is terrific +1. Haven't found any problems clicking around.

**Comment by michellephung** [comment]
+1

**Comment by garrensmith** [comment]
@robertkowalski I've added some more tests. Can you give a final review.

**Pull request merged**